src/components/Home.jsx

```diff
@@ -1,25 +1,74 @@
 import { fetchAllArticles, fetchArticlesByTopic } from "../api";
 import { useEffect, useState } from "react";
-import { useParams } from "react-router-dom";
+import { useParams, useSearchParams } from "react-router-dom";
 import { Link } from "react-router-dom";
 
 const Home = () => {
   const [allArticles, setAllArticles] = useState([]);
+  const [searchParams, setSearchParams] = useSearchParams({});
+  const [sortBy, setSortBy] = useState("created_at");
+  const [order, setOrder] = useState("DESC");
   const { topic_name } = useParams();
 
   useEffect(() => {
     if (!topic_name) {
-      fetchAllArticles().then((fetchedArticles) => {
+      fetchAllArticles(sortBy, order).then((fetchedArticles) => {
         setAllArticles(fetchedArticles.articles);
       });
     } else
       fetchArticlesByTopic(topic_name).then((topic) => {
         setAllArticles(topic.articles);
       });
-  }, [topic_name]);
+  }, [topic_name, sortBy, order]);
 
   return (
     <>
+      <section>
+        <span>Sort bar: </span>
+        <br />
+        <button
+          className="article_card--button"
+          onClick={() => {
+            setSearchParams({ sort_by: "created_at", order: order });
+            setSortBy("created_at");
+          }}
+        >
+          Date
+        </button>
+        <button
+          className="article_card--button"
+          onClick={() => {
+            setSearchParams({ sort_by: "comment_count", order: order });
+            setSortBy("comment_count");
+          }}
+        >
+          Comment Count
+        </button>
+        <button
+          className="article_card--button"
+          onClick={() => {
+            setSearchParams({ sort_by: "votes", order: order });
+            setSortBy("votes");
+          }}
+          value="votes"
+        >
+          Votes
+        </button>
+        <button
+          className="article_card--button"
+          onClick={() => {
+            setSearchParams({ sort_by: sortBy, order: order });
+            if (order === "DESC") {
```

Comment:
The URL and the order seem to get out of sync here, I think you will need to move the setSearchParams inside of the if/else logic and update that state accordingly